### PR TITLE
Added Taint and Tolerations support in crdb cluster #695

### LINKED ISSUE #695: Implement Taints and Tolerations
Need Taints and Tolerations:
In case we want only CRDB running on a class of nodes and the nodes are reserved for us

### PR DESCRIPTION
**Pull request by abhishekdwivedi3060**
Closes #695 
* Added Taint and Tolerations support in crdb cluster
* Added the new API field  "tolerations". tolerations is a []corev1.Toleration struct which allows the user to set tolerations for pod scheduling onto dedicated nodes
* Modified existing unit test for this change

### CODE REVIEW

**Review by chrislovecnm — CHANGES_REQUESTED** [review]
Looks good, if you can back out the changes to the files that are not needed in this PR please

File: apis/v1alpha1/cluster_types.go [hunk 1]
```diff
@@ -34,80 +34,103 @@ type CrdbClusterSpec struct {
 	// +operator-sdk:csv:customresourcedefinitions:type=spec,displayName="Number of nodes",xDescriptors="urn:alm:descriptor:com.tectonic.ui:podCount"
 	// +required
 	Nodes int32 `json:"nodes"`
+
```
[the remark this answers is not in the document]
Comment (abhishekdwivedi3060) [reply apis/v1alpha1/cluster_types.go]: Done, removed new lines

File: pkg/actor/actor.go [hunk 2]
```diff
@@ -18,6 +18,7 @@ package actor
 
 import (
 	"context"
+
```
[the remark this answers is not in the document]
Comment (abhishekdwivedi3060) [reply pkg/actor/actor.go]: Done

**Review by pseudomuto — COMMENTED** [review]
Code LGTM 👍 

Do we have a test to prove this does what we intend it to? I understand tolerations are a builtin, just wondering if we should update the e2e tests somehow to make sure this feature works.

**Comment by abhishekdwivedi3060** [comment]
There is unit-test to check if Tolerations field value is propagated to actual sts pod.
I don't think there is any e2e for existing fields like `AffinityRules`
As you mentioned these functionalities are built it, so not sure how useful that will be to add e2e for those

**Comment by pseudomuto** [comment]
> As you mentioned these functionalities are built it, so not sure how useful that will be to add e2e for those

Ok, that's kinda fair. I think we should at least test it though, even if it's manually. Just to be sure that it works, like that we set all the fields correctly and K8s does what we expect.

**Review by chrislovecnm — APPROVED** [review]

**Pull request merged**